src/Kestrel.Core/Internal/Http/HttpProtocol.SendFileFeature.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http
+{
+    public partial class HttpProtocol : IHttpSendFileFeature
+    {
+        async Task IHttpSendFileFeature.SendFileAsync(string path, long offset, long? count, CancellationToken cancellationToken)
+        {
+            var contentLength = ResponseHeaders.ContentLength;
+            if (contentLength.HasValue && count.HasValue && contentLength.Value != count.Value)
+            {
+                // throw
+            }
+
+            using (var fileStream = new FileStream(path,
+                                 FileMode.Open,
+                                 FileAccess.Read,
+                                 FileShare.ReadWrite,
+                                 8 * 4096, // Buffer size - not currently pooled
```

Comment:
We should use a super small buffer size here so we can avoid using the internal FileStreamBuffer.

Reply:
Is zero valid? 😄

Reply:
Nope 😄

Reply:
The checks are for <, so if you use a side of 1, the internal buffer won't be created.

Reply:
1 it is :)